src/marqo/tensor_search/tensor_search.py

```diff
@@ -2019,19 +2025,20 @@ def add_prefix_to_queries(queries: List[BulkSearchQueryEntity]) -> List[BulkSear
         if q.q is None:
             prefixed_q = q.q
         elif isinstance(q.q, str):
-            if _is_image(q.q):
-                prefixed_q = q.q
-            else:
+            modality = infer_modality(q.q, q.mediaDownloadHeaders)
```

Comment:
Is this the same logic? We don't want to change image modality inference logic

Reply:
Case 1: q is a string, then it will be categorized as text in both cases
Case 2: q is an valid URL, it will be categorized as an image in both cases
Case 3: q is an invalid URL, it will be categorized as an image before and raise an error during downloading. Now, we raise the error when inferring the modality.

Therefore I think the behaviour is the same.